pandas-stubs/_typing.pyi

```diff
@@ -427,12 +427,30 @@ ByT = TypeVar(
     Interval[Timedelta],
     tuple,
 )
+# Use a distinct SeriesByT when using groupby with Series of known dtype.
+# Essentially, an intersection between Series S1 TypeVar, and ByT TypeVar
+SeriesByT = TypeVar(
+    "SeriesByT",
+    str,
```

Comment:
We've been recently advised to no longer use constrained `TypeVars`, so can you change this to `SeriesByT = TypeVar("SeriesByT", bound=str | bytes | ...)`  We made the change for `S1` in a recent PR

Reply:
Hi,

What is the difference between `TypeVar("SeriesByT", bound = str | bytes ...)` and `TypeVar("SeriesByT, str, byts, ...)`  ?

I tried to replace the SeriesByT with the `bound` keyword, but then the mypy did not pass anymore :

```
Poe => mypy

===========================================
Beginning: 'Run mypy on 'tests' (using the local stubs) and on the local stubs'
===========================================

pandas-stubs/core/series.pyi:648: error: Type variable "SeriesByT" not valid as type argument value for "SeriesGroupBy"  [type-var]
pandas-stubs/core/frame.pyi:1100: error: Type variable "SeriesByT" not valid as type argument value for "DataFrameGroupBy"  [type-var]
Found 2 errors in 2 files (checked 224 source files)

===========================================
Step: 'Run mypy on 'tests' (using the local stubs) and on the local stubs' failed!
===========================================

```

Reply:
`bound` allows the type to be a union of all the listed types. Without it, the type has to be one of the listed (sub-)types.

You probably also need to adjust the type used for `SeriesGroupBy` and `DataFrameGroupBy` to use `bound`.

Reply:
Thanks I actually tried that and got rid of the errors.